src/deprecations.jl

```diff
@@ -57,3 +57,59 @@ Base.@deprecate_binding Line2D Line
 Base.@deprecate_binding Line3D Line
 Base.@deprecate_binding Quadrilateral3D Quadrilateral
 export Line2D, Line3D, Quadrilateral3D
+
+# Deprecation of auto-vectorized methods
+function add!(dh::DofHandler, name::Symbol, dim::Int)
+    celltype = getcelltype(dh.grid)
+    if !isconcretetype(celltype)
+        error("If you have more than one celltype in Grid, you must use add!(dh::DofHandler, fh::FieldHandler)")
+    end
+    Base.depwarn(
+        "`add!(dh::DofHandler, name::Symbol, dim::Int)` is deprecated. Instead, pass the " *
+        "interpolation explicitly, and vectorize it to `dim` for vector-valued " *
+        "fields. See CHANGELOG for more details.",
+        :add!,
+    )
+    ip = default_interpolation(celltype)
+    add!(dh, name, dim == 1 ? ip : VectorizedInterpolation{dim}(ip))
+end
+
+function add!(dh::DofHandler, name::Symbol, dim::Int, ip::ScalarInterpolation)
+    Base.depwarn(
+        "`add!(dh::DofHandler, name::Symbol, dim::Int, ip::ScalarInterpolation)` is " *
+        "deprecated. Instead, vectorize the interpolation to the appropriate dimension " *
+        "and add it (`vip = ip^dim; add!(dh, name, vip)`). See CHANGELOG for more details.",
```

Comment:
Maybe we can be more specific

```suggestion
        "and add it (`vip = ip^dim; add!(dh, name, vip)`). See CHANGELOG for version 0.4.0 for more details.",
```

Reply:
Same for the warning above and below.

Reply:
We can add a link instead (when that section of the CHANGELOG is written).